File: sample-super-rx-ai/src/main/java/com/mateuszkoslacz/moviper/rxsample/viper/presenter/FullscreenPhotoPresenter.java
package com.mateuszkoslacz.moviper.rxsample.viper.presenter;

import android.os.Bundle;
import android.support.annotation.NonNull;

import com.mateuszkoslacz.moviper.base.presenter.BaseRxPresenter;
import com.mateuszkoslacz.moviper.rxsample.viper.contract.FullscreenPhotoContract;
import com.mateuszkoslacz.moviper.rxsample.viper.interactor.FullscreenPhotoInteractor;
import com.mateuszkoslacz.moviper.rxsample.viper.routing.FullscreenPhotoRouting;
import com.mateuszkoslacz.moviper.rxsample.viper.view.activity.FullscreenPhotoActivity;

public class FullscreenPhotoPresenter
        extends BaseRxPresenter<FullscreenPhotoContract.View,
                                FullscreenPhotoContract.Interactor,
                                FullscreenPhotoContract.Routing>
        implements FullscreenPhotoContract.Presenter {

    private String mPhotoUrl;

    public FullscreenPhotoPresenter(Bundle bundle) {
        super(bundle);
        mPhotoUrl = getArgs().getString(FullscreenPhotoActivity.PHOTO_URL_EXTRA_STRING);
    }

    @Override
    public void attachView(FullscreenPhotoContract.View view) {
        super.attachView(view);
        if (isViewAttached()) getView().setPhoto(mPhotoUrl);
    }


    @NonNull
    @Override
    public FullscreenPhotoContract.Routing createRouting() {
        return new FullscreenPhotoRouting();
    }

    @NonNull
    @Override
    public FullscreenPhotoContract.Interactor createInteractor() {
        return new FullscreenPhotoInteractor();
    }
}

File: sample-super-rx-ai-kotlin/src/main/java/com/mateuszkoslacz/moviper/rxsample/viper/presenter/FullscreenPhotoPresenter.kt
package com.mateuszkoslacz.moviper.rxsample.viper.presenter

import android.os.Bundle

import com.mateuszkoslacz.moviper.base.presenter.BaseRxPresenter
import com.mateuszkoslacz.moviper.rxsample.viper.contract.FullscreenPhotoContract
import com.mateuszkoslacz.moviper.rxsample.viper.interactor.FullscreenPhotoInteractor
import com.mateuszkoslacz.moviper.rxsample.viper.routing.FullscreenPhotoRouting
import com.mateuszkoslacz.moviper.rxsample.viper.view.activity.FullscreenPhotoActivity

class FullscreenPhotoPresenter(bundle: Bundle) :
        BaseRxPresenter<FullscreenPhotoContract.View,
                FullscreenPhotoContract.Interactor,
                FullscreenPhotoContract.Routing>(bundle),
        FullscreenPhotoContract.Presenter {

    private val mPhotoUrl: String = args.getString(FullscreenPhotoActivity.PHOTO_URL_EXTRA_STRING)

    override fun attachView(view: FullscreenPhotoContract.View) {
        super.attachView(view)
        getView()?.setPhoto(mPhotoUrl)
    }

    override fun createRouting(): FullscreenPhotoContract.Routing = FullscreenPhotoRouting()

    override fun createInteractor(): FullscreenPhotoContract.Interactor = FullscreenPhotoInteractor()
}
